src/libsystemd/sd-bus/bus-introspect.c

```diff
@@ -101,6 +101,12 @@ static int introspect_write_arguments(struct introspect *i, const char *signatur
 
                 fprintf(i->f, "   <arg type=\"%.*s\"", (int) l, signature);
 
+                if(*names) {
+                    int ret = fprintf(i->f, " name=\"%s\"", names)-8;
+                    if(ret>0)
+                        names += ret+1;
+                }
```

Comment:
Please follow our coding style:
- add a whitespace between `if` and `(`.
- also, `ret > 0`, that is add whitespaces before and after `>`.
- use 8ch indentation.

Why the first character `*name` is checked. How about the following?
```
if (names)
        fprintf(...);
```
And drop `strempty()` below.

`names += ret + 1`: Why do you need this? It seems do nothing.

Reply:
ok for the changes in the format, I'll do it. I will also change ret to len.
the algorithm is however ok as it is. 'names' contains multiple names for the introspectable xml, separated by \0 as per Lennart Poettering suggestions on the mailing list.

Reply:
Oops, I missed the logic is in `for` loop. Sorry.

Reply:
it should be fixed now. It compiles and run on my machine.

Reply:
Indentation is not updated yet...

Reply:
oops, I missed completely that, sorry. fixed.
and thanks for the review.

Reply:
Frankly, using the return value of `fprintf` to get the length of name is a bit too convoluted...

I guess I'd prefer the simpler, more direct:

```c
if (*names) {
        fprintf(i->f, " name=\"%s\"", names);
        names += strlen(names) + 1;
}
```

Yes, this ends up traversing `names` one more time to calculate its length, but that shouldn't be too much of a concern here...

Reply:
that's how I would have coded it as well. But I have received comments in other projects that I should avoid it re-traversing the strings in any case, so I have submitted this version.
No big deal I suppose. Shall I change it?

Reply:
ok, fine with me. I changed it.